gems/sorbet-runtime/lib/types/enum.rb

```diff
@@ -72,13 +72,16 @@ def self.each_value(&blk)
   #
   # Note: It would have been nice to make this method final before people started overriding it.
   # Note: Failed CriticalMethodsNoRuntimeTypingTest
-  sig {params(serialized_val: SerializedVal).returns(T.nilable(T.attached_class)).checked(:never)}
-  def self.try_deserialize(serialized_val)
+  sig do
+    params(serialized_val: SerializedVal, default_value: T.nilable(T::Enum))
+    .returns(T.nilable(T.any(T::Enum, T.attached_class))).checked(:never)
+  end
+  def self.try_deserialize(serialized_val, default_value: nil)
```

Comment:
1. You used an optional keyword arg instead of an optional positional arg.
2. Your return type is too wide. Even for people that don't use the new `default_value` parameter, it will break their code by forcing them to consider the case where it's a `T::Enum` not a `MyEnum`. This can be fixed with a generic.

```suggestion
  sig do
    type_parameters(:U)
      .params(
        serialized_val: SerializedVal,
        default_value: T.nilable(T.type_parameter(:U))
      )
      .returns(T.nilable(T.any(T.attached_class, T.type_parameter(:U))))
      .checked(:never)
  end
  def self.try_deserialize(serialized_val, default_value: nil)
```